ledger/acctupdates.go

```diff
@@ -1473,6 +1483,72 @@ func (au *accountUpdates) upgradeDatabaseSchema3(ctx context.Context, tx *sql.Tx
 	return 4, nil
 }
 
+
+// upgradeDatabaseSchema4 upgrades the database schema from version 4 to version 5,
+// cleaning old empty catchpoint directories.
+func (au *accountUpdates) upgradeDatabaseSchema4(ctx context.Context, tx *sql.Tx) (updatedDBVersion int32, err error) {
+	//validate no empty dirs
+	err = removeEmptyDirsOnSchemaUpgrade()
+	if err != nil {
+		return 0, err
+	}
+
+	// update version
+	_, err = db.SetUserVersion(ctx, tx, 5)
+	if err != nil {
+		return 0, fmt.Errorf("accountsInitialize unable to update database schema version from 4 to 5: %v", err)
+	}
+	return 5, nil
+}
+
+func  removeEmptyDirsOnSchemaUpgrade() (err error) {
+	f, err := os.Open(catchpointDirName)
+	isNotExists := os.IsNotExist(err)
+	f.Close()
+	if isNotExists {
+		return nil
+	}
+	for {
```

Comment:
Why do we need this for-loop ?

Reply:
It seems like we don't want to load the full list of catchpoint files into memory, So we execute the loop in chunks.